plugins/generator-1.21.1/neoforge-1.21.1/templates/gui/gui_msg_menustate.java.ftl

```diff
@@ -0,0 +1,93 @@
+<#--
+ # MCreator (https://mcreator.net/)
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2025, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+<#include "../procedures.java.ftl">
+
+package ${package}.network;
+
+@EventBusSubscriber(bus = EventBusSubscriber.Bus.MOD) public record ${JavaModName}MenustateUpdateMessage(int elementType, String name, Object elementState) implements CustomPacketPayload {
+
+	public static final Type<${JavaModName}MenustateUpdateMessage> TYPE = new Type<>(ResourceLocation.fromNamespaceAndPath(${JavaModName}.MODID, "guistate_update"));
+
+	public static final StreamCodec<RegistryFriendlyByteBuf, ${JavaModName}MenustateUpdateMessage> STREAM_CODEC = StreamCodec.of(${JavaModName}MenustateUpdateMessage::write, ${JavaModName}MenustateUpdateMessage::read);
+
+	public static void write(FriendlyByteBuf buffer, ${JavaModName}MenustateUpdateMessage message) {
+		int eType = message.elementType;

```

Comment:
I recommend using full name elementType for readability, it is static method so this does not collide with record's elementType. Apples to both versions